core/templates/pages/exploration-player-page/services/current-interaction.service.spec.ts

```diff
@@ -114,4 +126,36 @@ describe('Current Interaction Service', () => {
     expect(hookStateA).toEqual(1);
     expect(hookStateB).toEqual(3);
   });
+
+  it('should throw error on submitting when submitAnswerFn is null', () => {
+    spyOn(playerPositionService, 'getDisplayedCardIndex').and.returnValue(1);
+    spyOn(playerTranscriptService, 'getCard').and.returnValue(
+      StateCard.createNewCard(
+        'First State', 'Content HTML',
+        '<oppia-text-input-html></oppia-text-input-html>',
+        null, null, null, '', null));
+    spyOn(contextService, 'getExplorationId').and.returnValue('abc');
+    spyOn(contextService, 'getPageContext').and.returnValue('learner');
+
+    let additionalInfo = (
+      '\nUndefined submit answer debug logs:' +
+      '\nInteraction ID: null' +
+      '\nExploration ID: abc' +
+      '\nState Name: First State' +
+      '\nContext: learner' +
+      '\nErrored at index: 1');
+
+    currentInteractionService.registerCurrentInteraction(null, null);
+
+    expect(() => currentInteractionService.submitAnswer()).toThrowError(
+      'The current interaction did not ' + 'register a _submitAnswerFn.' +
+        additionalInfo);
+  });
+
+  it('should update view with new answer', () => {
+    expect(currentInteractionService.updateViewWithNewAnswer).toBeDefined();
```

Comment:
Can we test with a value instead of `toBeDefined`?

Reply:
In this line, I am just asserting that `updateViewWithNewAnswer` function is defined. And in the next line, `toBeDefined` is used instead of a value because `onAnswerChanged$` returns the observable of `answerChangedSubject` which is a private static value of the class. Please let me know if I am missing something here

Reply:
Please Add a comment in the code explaining the same

Reply:
done @gp201